### ISSUE TITLE
New Copyedit System

### ISSUE THREAD

**Issue by MartinPaulEve** [BirkbeckCTP/janeway#1420]
**Is your feature request related to a problem? Please describe.**
Copyediting is a messy process. As we are dealing with many actors working with single-locking files, we often resort to email. The process is also more conversational than the linear workflow suggests. As such, the rigid confinement to an end-to-end workflow is not really always the best way to handle this.

**Describe the solution you'd like**
It would be better to have a potential workflow path, but also a more freeform system.

Ideally, actors would check out files when they are working on it and check the files back in when done, preventing others from editing the file at that time. We'd want a "working master" document at all times, but to preserve a history log of all past files and have the ability to restore files from the history.

There could be certain modes here:

1. Free for all -- anyone can check in and check out
2. Workflow -- sequential assignation

There should also be some kind of chat/thread facility for discussions around what is happening.

Finally, if a deadline is missed, the file should automatically be checked in, with a log note that this is what has happened. The manager should also always be able to wrest back control of the workflow too/force a check-in.

**Describe alternatives you've considered**
Massive painful email chains.

**Additional context**
As discussed on Discord.

**Comment by ajrbyers** [comment]
For clarification 2. would be our existing workflow right?

**Comment by MartinPaulEve** [comment]
I think so, yes

**Comment by rhb123** [comment]
Just to note that this would be very helpful for version 1.4 if possible - thanks

**Comment by hinefuku** [comment]
Would be useful to also make it easier for copyeditors to communicate directly with authors through system

**Comment by ajrbyers** [comment]
Yes this forms part of the forum suggestions you made previously, @hinefuku

**Comment by mauromsl** [comment]
A requirement from Simon at ZFF:
After a Copyedit assignment has been completed, further copyedit assignments should allow choosing the previous copyedit file as the target for the assignment. They would like to be able to pass on the copyedited file onto  a second copyeditor (at the moment, the copyedit assignment screen only allows to pick manuscript files, not copyedit files, so they are re-uploading these as manuscripts as a workaround)

**Comment by mauromsl** [comment]
A suggestion from JCMS ediitors:
For smaller teams where the editors are responsible for the copyediting, they would like to be able to upload the copyedited manuscript directly, instead of having to create a task for themselves. (They can currently do this for the typesetting stage, but not for copyediting)

**Comment by mauromsl** [comment]
Another problem with the current implementation:
Once a copyedit task has been created wit a given copyeditor, they are no longer selectable in future tasks. It is also not possible for editors to reopen a task making it imposible for requesting further changes to the same copyeditor.

**Comment by severett89** [comment]
It is possible that authors may struggle to find their copyedit review task from the dashboard in Janeway: our email template doesn't have a direct link to the review task and just goes to dashboard. We also don't have a deadline date to set for review tasks in copyedit and have to manually insert one. A date picker and deadline would be handy here.

**Comment by joemull** [comment]
Other functional requirements for new system
* Let someone other than author review copyedits (implied in check-out system described by Martin)

**Comment by pgoussy** [comment]
Thoughts from Michigan Publishing Services:

From our perspective, the workflow for assigning a copyedited manuscript for author review is effectively identical to assigning a typeset PDF/rendered HTML for author review. Therefore, in the interest of standardization, we'd like to see a new copyediting system that directly mirrors the UI and functionality of the Typesetting Plugin (in terms of selecting assignees, due dates, files, etc).

NB: to date, MPS' typesetters have _not_ been working in Janeway, so we've needed to upload the typeset files ourselves before creating proofreading assignments. This means that we haven't been using the Janeway interface for the initial typeset, only the reviews of typeset files. 

So, perhaps it would be more accurate to say that we want to see the _copyedit author reviews_ mirror the UI/functionality of the _author proofreading assignments_ in the Typesetting Plugin. We can't comment strongly on whether the UI for copyediting assignments should directly mirror typesetting assignments, since we don't really have experience with the latter.

**Comment by ajrbyers** [comment]
Feedback also here https://github.com/BirkbeckCTP/janeway/issues/1411
